src/test/java/org/elasticsearch/index/engine/internal/InternalEngineTests.java

```diff
@@ -1559,4 +1562,78 @@ public void testSettings() {
 
         }
     }
+
+    @Test
+    public void testRetryWithAutogeneratedIdWorksAndNoDuplicateDocs() throws IOException {
+
+        ParsedDocument doc = testParsedDocument("1", "1", "test", null, -1, -1, testDocument(), B_1, false);
+        boolean canHaveDuplicates = false;
```

Comment:
Oh, I see. you have two tests. We can maybe fold it into one and have the canHaveDuplicates set randomly and then set it at the reverse?